barquesubmit/client.go

```diff
@@ -0,0 +1,203 @@
+package barquesubmit
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"path"
+	"strings"
+
+	"github.com/evergreen-ci/gimlet"
+	"github.com/evergreen-ci/utility"
+	"github.com/mongodb/amboy"
+	"github.com/mongodb/curator/repobuilder"
+	"github.com/pkg/errors"
+)
+
+const (
+	barqueAPIKeyHeader  = "Api-Key"
+	barqueAPIUserHeader = "Api-User"
+)
+
+type Client struct {
+	baseURL  string
+	username string
+	apiKey   string
+}
+
+func New(baseURL string) (*Client, error) {
+	if !strings.HasPrefix(baseURL, "http") {
+		return nil, errors.New("malformed url")
+	}
+
+	if !strings.HasSuffix(baseURL, "/") {
+		baseURL += "/"
+	}
+
+	if !strings.HasSuffix(baseURL, "/rest/v1/") {
+		baseURL += "rest/v1/"
+	}
+
+	return &Client{
+		baseURL: baseURL,
+	}, nil
+}
+
+func (c *Client) getURL(p string) string {
+	if strings.HasPrefix(p, c.baseURL) {
+		return p
+	}
+
+	return path.Join(c.baseURL, p)
+}
+
+func (c *Client) makeRequest(ctx context.Context, url, method string, body io.Reader) (*http.Request, error) {
+	req, err := http.NewRequest(method, c.getURL(url), body)
+	if err != nil {
+		return nil, errors.Wrap(err, "problem with request")
+	}
+	req = req.WithContext(ctx)
+
+	if c.apiKey == "" {
```

Comment:
I see why you did this, but it seems slightly silly. I think you can remove.

Reply:
eh?